src/gmm_utils.erl

```diff
@@ -0,0 +1,94 @@
+%%%-------------------------------------------------------------------
+%% @doc
+%%  Basic functions to switch between JSON and erlang types
+%% @end
+%%%-------------------------------------------------------------------
+
+-module(gmm_utils).
+
+%% API
+-export([
+    empty_json/0,
+    encode/1,
+    decode/1,
+    create_vertex_id/1,
+    create_vertex_id/2,
+    split_bin/1,
+    parse_boolean/1,
+    convert_microseconds_to_iso_8601/1
+]).
+
+-include("records.hrl").
+
+%%%---------------------------
+%% Implementations
+%%%---------------------------
+
+%% JSON manipulation
+-spec empty_json() -> binary().
+empty_json() -> jiffy:encode({[]}).
+
+-spec encode(any()) -> binary().
+encode(Val) -> jiffy:encode(Val).
+
+-spec decode(binary()) -> any().
+decode(Val) -> jiffy:decode(Val, [return_maps]).
+
+%% Binary string manipulation
+-spec create_vertex_id(Name :: binary()) -> binary().
+create_vertex_id(Name) ->
+    create_vertex_id(Name, list_to_binary(?ZONE_ID)).
+
+-spec create_vertex_id(Name :: binary(), Zone :: binary()) -> binary().
+create_vertex_id(Name, Zone) ->
+    <<Zone/binary, "/", Name/binary>>.
+
+-spec split_bin(Bin :: binary()) -> {ok, list(binary())} | {error, any()}.
+split_bin(Bin) ->
+    %% default delimiter is '/'
```

Comment:
Widząc taki komentarz aż nie mogę się powstrzymać. Skoro jest opcja domyślna, to powinna być możliwość zmiany jej wartości